actors/builtin/power/power_actor.go

```diff
@@ -192,116 +192,125 @@ func (a Actor) DeleteMiner(rt Runtime, params *DeleteMinerParams) *adt.EmptyValu
 }
 
 type OnSectorProveCommitParams struct {
-	Weight SectorStorageWeightDesc
+	RawBytePower         abi.StoragePower
+	QualityAdjustedPower abi.StoragePower
 }
 
-// Returns the initial pledge collateral requirement.
+// Adds power for the calling actor and returns the initial pledge requirement for committing that
+// much power at this epoch.
+// May only be invoked by a miner actor.
 func (a Actor) OnSectorProveCommit(rt Runtime, params *OnSectorProveCommitParams) *abi.TokenAmount {
 	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	initialPledge := a.computeInitialPledge(rt, &params.Weight)
+	builtin.RequireParam(rt, params.RawBytePower.GreaterThanEqual(big.Zero()), "negative raw power %s", params.RawBytePower)
```

Comment:
Question: I'm curious why these are being added. I get that positive power is an invariant, but it's a weak one. The real invariant is that the power must be correctly computed, and for that the power actor is dependent on the miner.

Is it to catch bugs in the miners calculation? If so, should it happen on the miner side where there is more context?
Is it to prevent improper usage? I can see the potential for the method to be misused to remove power.

This will set a precedent for future assertions in actor code so I'd like to understand the reasoning.

Reply:
Right, this is all cleaned up in #487, where they are removed again for a method that is actually expecting negative values. Yes, it was to catch bugs or improper usage on the miner side.